types/src/transaction/mod.rs

```diff
@@ -635,6 +635,27 @@ impl From<VMStatus> for TransactionStatus {
     }
 }
 
+/// The result of running the transaction through the VM validator.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VMValidatorResult {
+    status: Option<VMStatus>,
```

Comment:
If this is `None`, will the mempool ever read `gas_price`?

Reply:
yes, mempool will use gas priority value. None just means there's no vm error